### PR TITLE
feat(fa): Creating a printable application with files

### PR DESCRIPTION
**Pull request by MargretFinnboga**
# Creating a printable application with files

https://app.asana.com/0/1201927126329736/1201445348663381

## What

- Make a page where application is printable 
- Fetch all image and display them 
- Fetch all pdfs and display them as well 
- Update pdf viewer when you want to display whole pdf and as canvas 

## Why

- Requested from staff in municipality 
- To use pdf viewer in more ways

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

application
Screenshot 2022-03-22 at 10 14 48

printable view

Screenshot 2022-03-22 at 10 15 10

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: apps/financial-aid/web-veita/src/components/PrintableFiles/PrintableFiles.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      variables: { input: { id: el.id } },
+      fetchPolicy: 'no-cache',
+      errorPolicy: 'all',
+    })
+    if (data?.getSignedUrlForId) {
+      allIPdfs.push(data.getSignedUrlForId.url)
+    }
+  })
+
+  if (allIPdfs.length === files.length) {
```
Comment (addi) [inline apps/financial-aid/web-veita/src/components/PrintableFiles/PrintableFiles.tsx:26]: Það er hægt að gera betri error logic in þetta. Þetta er frekar weird.

File: apps/financial-aid/web-veita/src/components/PrintableFiles/PrintableFiles.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+interface Props {
+  files: ApplicationFile[]
+  isImages?: boolean
+}
+
+const PrintableFiles = ({ files, isImages = false }: Props) => {
+  const allIPdfs: string[] = []
+
+  files.map((el) => {
+    const { data } = useQuery(GetSignedUrlQuery, {
```
Comment (addi) [inline apps/financial-aid/web-veita/src/components/PrintableFiles/PrintableFiles.tsx:16]: Já spurning um að gera endapunkt sem skilar signed url yfir allar skrár.

File: apps/financial-aid/web-veita/src/components/PrintableFiles/PrintableFiles.tsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+import { Box, Button, PdfViewer } from '@island.is/island-ui/core'
+import { GetSignedUrlQuery } from '@island.is/financial-aid-web/veita/graphql'
+
+interface Props {
+  files: ApplicationFile[]
+  isImages?: boolean
+}
+
+const PrintableFiles = ({ files, isImages = false }: Props) => {
+  const allIPdfs: string[] = []
```
Comment (addi) [inline apps/financial-aid/web-veita/src/components/PrintableFiles/PrintableFiles.tsx:13]: Eru ekki bara pdf, eru ekki líka myndir hérna.

File: apps/financial-aid/web-veita/src/routes/Application/application.tsx [hunk 4]
```diff
@@ -18,7 +18,7 @@ interface ApplicantData {
   application: Application
 }
 
-const UserApplication = () => {
+const PrintableApplication = () => {
   const router = useRouter()
 
   const [isLoading, setIsLoading] = useState(false)
```
Comment (addi) [inline apps/financial-aid/web-veita/src/routes/Application/application.tsx:24]: Þarftu þennan? er ekki nóg að hafa loading sem er þarna fyrir neðan?

**Review by addi — CHANGES_REQUESTED** [review]

File: apps/financial-aid/web-veita/src/routes/Application/application.tsx (hunk 4 above)
Comment (MargretFinnboga) [reply apps/financial-aid/web-veita/src/routes/Application/application.tsx:24]: Nei þarf þennan þegar application er uppfært 😇

Comment (MargretFinnboga) [reply apps/financial-aid/web-veita/src/routes/Application/application.tsx:24]: En kannski óþarfi þarna þar sem ekki er hægt að ýta á takka 🙃

File: libs/island-ui/core/src/lib/PdfViewer/PdfViewer.tsx [hunk 5]
```diff
@@ -65,27 +71,37 @@ export const PdfViewer: FC<PdfViewerProps> = ({ file }) => {
...
           onLoadSuccess={onDocumentLoadSuccess}
-          renderMode="svg"
+          renderMode={renderMode}
           className={styles.pdfViewer}
           loading={() => loadingView()}
         >
-          <pdfLib.Page pageNumber={pageNumber} />
+          {showAllPages ? (
+            <>
+              {Array.from(new Array(numPages), (el, index) => (
```
[the remark this answers is not in the document]
Comment (MargretFinnboga) [reply libs/island-ui/core/src/lib/PdfViewer/PdfViewer.tsx]: Array.from(new Array(numPages), þá þetta eða tjékið?

File: apps/financial-aid/web-veita/src/components/AdminLayout/AdminLayout.tsx [hunk 6]
```diff
@@ -34,20 +34,29 @@ const AdminLayout = ({ children, className }: PageProps) => {
     }
   }, [showNavMobile])
 
+  const printPage = router.pathname.includes('print')
```
[the remark this answers is not in the document]
Comment (MargretFinnboga) [reply apps/financial-aid/web-veita/src/components/AdminLayout/AdminLayout.tsx]: alvöru route hvað meinaru 🤓 ?

**Review by Arnarkari93 — APPROVED** [review]
LGTM from the `island-ui` perspective

**Review by davidpxn — APPROVED** [review]
This looks good to me! 👏

**Pull request merged**